crates/router/src/core/payment_methods/vault.rs

```diff
@@ -0,0 +1,561 @@
+        payment_method: &api::PaymentMethod,
+        customer_id: Option<String>,
+    ) -> RouterResult<String> {
+        let value1 = payment_method
+            .get_value1(customer_id.clone())
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Error getting Value1 for locker")?;
+
+        let value2 = payment_method
+            .get_value2(customer_id)
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Error getting Value12 for locker")?;
+
+        let lookup_key = token_id.unwrap_or_else(|| generate_id_with_default_len("token"));
+
+        let db_value = MockTokenizeDBValue { value1, value2 };
+
+        let value_string =
+            utils::Encode::<MockTokenizeDBValue>::encode_to_string_of_json(&db_value)
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Failed to encode payment method as mock tokenize db value")?;
+
+        let config = storage::ConfigNew {
+            key: lookup_key.clone(),
+            config: value_string,
+        };
+
+        state
+            .store
+            .insert_config(config)
+            .await
+            .change_context(errors::ApiErrorResponse::InternalServerError)
+            .attach_printable("Mock tokenization save to db failed")?;
+
+        Ok(lookup_key)
+    }
+
+    #[instrument(skip_all)]
+    pub async fn delete_locker_payment_method_by_lookup_key(
+        state: &routes::AppState,
+        lookup_key: &Option<String>,
+    ) {
+        let db = &*state.store;
+        if let Some(id) = lookup_key {
+            match db.delete_config_by_key(id).await {
+                Ok(_) => logger::info!("Card Deleted from locker mock up"),
+                Err(err) => logger::error!("Err: Card Delete from locker Failed : {}", err),
+            }
+        }
+    }
+}
+
+#[cfg(feature = "basilisk")]
+impl Vault {
+    #[instrument(skip_all)]
+    pub async fn get_payment_method_data_from_locker(
+        state: &routes::AppState,
+        lookup_key: &str,
+    ) -> RouterResult<(Option<api::PaymentMethod>, SupplementaryVaultData)> {
+        let de_tokenize = get_tokenized_data(state, lookup_key, true).await?;
+        let (payment_method, customer_id) =
+            api::PaymentMethod::from_values(de_tokenize.value1, de_tokenize.value2)
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("Error parsing Payment Method from Values")?;
+
+        Ok((Some(payment_method), customer_id))
+    }
+
+    #[instrument(skip_all)]
+    pub async fn store_payment_method_data_in_locker(
+        state: &routes::AppState,
+        token_id: Option<String>,
+        payment_method: &api::PaymentMethod,
+        customer_id: Option<String>,
+    ) -> RouterResult<String> {
+        let value1 = payment_method
+            .get_value1(customer_id.clone())
```

Comment:
Can we do a implicit conversion of  `InternalServerError` in `get_value1` function itself.
Or change_change_context automatically lifting into `InternalServerError`.

So code might look like:

```
let value1 = payment_method.get_value1(&customer_id)?;
let value2 = payment_method.get_value2(&customer_id)?;
let lookup_key = token_id.unwrap_or_else(|| generate_id_with_default_len("token"));

create_tokenize(state, value1, Some(value2), lookup_key).await
```

Reply:
Ok, we take this up later